### ISSUE TITLE
Highlighting on server?

### ISSUE THREAD

**Issue by Pierrci** [rtfpessoa/diff2html#345]
Hi,

I'm currently experimenting with diff2html server-side, and I was wondering if it was possible to highlight the code with highligth.js from there? All the examples I've seen for highlighting the code are client-side, and my different tests were unsuccessful so far.

Thanks!

**Comment by rtfpessoa** [comment]
Hi @Pierrci,

I never tried doing highlight in server side.

The code relies in the dom, and uses several operations in it to mutate the dom.

You can check all the code here https://github.com/rtfpessoa/diff2html/blob/master/src/ui/js/diff2html-ui-base.ts#L111

In this case you could probably re-implement this with jsdom or something similar, should not be super hard, but I never tried.

Can you explain a bit more of your use case so I can understand what you are trying to do?

I have the [diff2html-cli](https://github.com/rtfpessoa/diff2html-cli) project renders the html in the server but then when you open the html it will highlight after. Would this type of usage work for you?

**Comment by Pierrci** [comment]
We're using handlebars to compile templates from the server, and we would like being able to directly insert the highlighted version of the diff in the template, but I understand the limitations server-side since it needs the dom.

In parallel, since `Diff2HtmlUI` accepts directly an element with a diff in it, I've been trying a similar behavior to the one of the CLI you linked to, and might go this way if we can't overcome server-side limitations. In my tests though, I've been unable to import the `Diff2HtmlUI` class directly from a TS file with something like `import { Diff2HtmlUI } from 'diff2html'` (on the front side), and I had to fall back to importing it with a `<script>` tag. Is there a way to avoid that? (I can open another issue for that if you prefer).

**Comment by rtfpessoa** [comment]
Diff2HtmlUI is not exported in the root, you need to import it directly from the path `import { Diff2HtmlUI } from 'diff2html/lib/ui/js/diff2html-ui'`.

I have a complete list under distributions in https://github.com/rtfpessoa/diff2html#distributions

**Comment by rtfpessoa** [comment]
Closing due to inactivity
